awx/main/tasks.py

```diff
@@ -3251,10 +3256,21 @@ def work_type(self):
 
     @cleanup_new_process
     def cancel_watcher(self, processor_future):
+        receptor_ctl = get_receptor_ctl()
         while True:
             if processor_future.done():
                 return processor_future.result()
 
+            # cancel job if receptor no longer knows about work item
+            try:
+                receptor_ctl.simple_command(f'work status {self.unit_id}')
+            except RuntimeError:
+                self.task.instance.result_traceback = traceback.format_exc()
+                self.task.instance.save(update_fields=['result_traceback'])
+
+                result = namedtuple('result', ['status', 'rc'])
+                return result('error', 1)
```

Comment:
We should either remove or revert this commit, because it was reverted downstream and git just messed it up somehow.

Reply:
do we need to revert again downstream?

Reply:
Revert the whole commit, or just this line of code?

Reply:
the whole commit 2a11bb4f3bc41feab9c563e2c653f44fb365a8a2. Doesn't matter to me how you do the git process.